src/runtime/security/security.thrift

```diff
@@ -0,0 +1,46 @@
+include "../../dsn.thrift"
+
+namespace cpp dsn.security
+
+// negotiation process:
+//
+//                       client                              server
+//                          | ---   SASL_LIST_MECHANISMS    --> |
+//                          | <-- SASL_LIST_MECHANISMS_RESP --- |
+//                          | --    SASL_SELECT_MECHANISMS  --> |
+//                          | <-- SASL_SELECT_MECHANISMS_OK --- |
+//                          |                                   |
+//                          | ---      SASL_INITIATE        --> |
+//                          |                                   |
+//                          | <--      SASL_CHALLENGE       --- |
+//                          | ---  SASL_CHALLENGE_RESPONSE  --> |
```

Comment:
How does the server send request to the client? Does that mean the client should open a fixed port listening to server request? How do you plan to implement this round of RPC?

Reply:
The client must first start the negotiation, and server rpc session will wait for the negotiation message from client. There is no need to open another listening port, just reuse the rpc session port.
If a server rpc session receive message, it should judge whether it is negotiation message or not. It it is negotiation message, let the member of `negotation`to deal with it.